src/dex-ui/pages/ProposalDetails/useProposalDetails.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+  useDexContext,
+  useHasVoted,
+  useCastVote,
+  useExecuteProposal,
+  useProposal,
+  useClaimGODTokens,
+} from "../../hooks";
+import { ProposalState, ProposalStatus } from "../../store/governanceSlice";
+import { createHashScanLink, getStatusColor } from "../../utils";
+
+export function useProposalDetails(proposalId: string | undefined) {
```

Comment:
@mgarbs Here is an example of a Hook. It allows us to extract out server-state, client-state, other side-effects, formatting, conditional logic, etc. out of the main UI component file. We're calling this hook in the `[ProposalDetails](https://github.com/hashgraph/hedera-accelerator-defi-dex-ui/pull/79/files#diff-f8115b6cf267daf061ab26a62850ad8560ad002375aedeaa7f5dc7a6825637e6)` component. 

Hooks are standard JS function except they can interact with the React component lifecycle. For example, you can use a `useState()` or `useEffect` inside of a React hook but not inside of a standard JS function. Hooks can be comprised of multiple other hooks - we could technically break out some of the logic below into additional hooks to improve composability/readability.

Reply:
Very helpful, thank you